### ISSUE TITLE
[Bug] No EnumOwnedObjects in SMO for Powershell 7.1

### ISSUE THREAD

**Issue by Zhunya** [microsoft/sqlmanagementobjects#48]
Got an error when running dbatools in PowerShell 7.1, in PowerShell 5.1 all works just fine.

This happens when we try to call EnumOwnedObjects() on object [Microsoft.SqlServer.Management.Smo.User]

https://github.com/sqlcollaborative/dbatools/blob/a0dfe9773a38ae480f0f2c0a02e9735cfa7d5d74/functions/Remove-DbaDbUser.ps1#L105-L116

```powershell

            foreach ($user in $users) {
                $db = $user.Parent
                $server = $db.Parent
                $ownedObjects = $false
                $alterSchemas = @()
                $dropSchemas = @()
                Write-Message -Level Verbose -Message "Removing User $user from Database $db on target $server"
                if ($Pscmdlet.ShouldProcess($user, "Removing user from Database $db")) {
                    # Drop Schemas owned by the user before dropping the user
                    $schemaUrns = $user.EnumOwnedObjects() | Where-Object Type -EQ Schema
                    if ($schemaUrns) {
```

```powershell
PSMessageDetails      :
Exception             : System.Management.Automation.MethodInvocationException: Exception calling "EnumOwnedObjects" with "0" argument(s): "Constructor
                        on type 'Microsoft.SqlServer.Management.Smo.SqlObjectUnion' not found."
                         ---> System.MissingMethodException: Constructor on type 'Microsoft.SqlServer.Management.Smo.SqlObjectUnion' not found.
                           at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture)
                           at Microsoft.SqlServer.Management.Sdk.Sfc.ObjectCache.CreateObjectInstance(Assembly assembly, String objectType)
                           at Microsoft.SqlServer.Management.Sdk.Sfc.ObjectCache.LoadElement(ObjectLoadInfo oli)
                           at Microsoft.SqlServer.Management.Sdk.Sfc.ObjectCache.LoadElement(ObjectLoadInfo oli, ServerVersion ver, DatabaseEngineType d
                        atabaseEngineType, DatabaseEngineEdition databaseEngineEdition)
                           at Microsoft.SqlServer.Management.Sdk.Sfc.ObjectCache.GetElement(ObjectLoadInfo oli, ServerVersion ver, DatabaseEngineType da
                        tabaseEngineType, DatabaseEngineEdition databaseEngineEdition)
                           at Microsoft.SqlServer.Management.Sdk.Sfc.ObjectCache.GetAllElements(Urn urn, ServerVersion ver, DatabaseEngineType databaseE
                        ngineType, DatabaseEngineEdition databaseEngineEdition, Object ci)
                           at Microsoft.SqlServer.Management.Sdk.Sfc.Environment.GetObjectsFromCache(Urn urn, Object ci)
                           at Microsoft.SqlServer.Management.Sdk.Sfc.Environment.GetData(Request req, Object ci)
                           at Microsoft.SqlServer.Management.Sdk.Sfc.Enumerator.GetData(Object connectionInfo, Request request)
                           at Microsoft.SqlServer.Management.Smo.ExecutionManager.GetEnumeratorDataReader(Request req)
                           at Microsoft.SqlServer.Management.Smo.PermissionWorker.EnumOwnedObjects(SqlSmoObject obj)
                           at CallSite.Target(Closure , CallSite , Object )
                           --- End of inner exception stack trace ---
                           at System.Management.Automation.ExceptionHandlingOps.ConvertToMethodInvocationException(Exception exception, Type typeToThrow
                        , String methodName, Int32 numArgs, MemberInfo memberInfo)
                           at CallSite.Target(Closure , CallSite , Object )
                           at System.Management.Automation.Interpreter.DynamicInstruction`2.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
                           at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
TargetObject          :
CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
FullyQualifiedErrorId : MissingMethodException
ErrorDetails          :
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Remove-DbUser, E:\GitHub\sqlcollaborative\functions\Remove-DbaDbUser.ps1: line 115
                        at Remove-DbaDbUser<Process>, E:\GitHub\sqlcollaborative\functions\Remove-DbaDbUser.ps1: line 225
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}
```

**Comment by shueybubbles** [comment]
hmm.
ObjectCache.CreateObjetInstance just does this

```
            return assembly.CreateInstance(objectType,
                        false, BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.CreateInstance,
                        null, null, System.Globalization.CultureInfo.InvariantCulture, null);
```
SqlObjectUnion is in SqlEnum, declared as an internal class:

```
internal class SqlObjectUnion : SqlObject
	{
		ArrayList m_listObjects;

		SqlObjectUnion()
		{
			m_listObjects = new ArrayList();
		}
```

Perhaps the assembly it's trying to load from is wrong? Will investigate.

**Comment by shueybubbles** [comment]
@Zhunya  what version of SMO are you using? I think this was fixed in 161.40241.8 and newer .

**Comment by Zhunya** [comment]
Thank you, I'll ask the team about updating SMO to the latest version, current in a module is - 15.0.18218.0
